lib/engine/game/g_1880/step/buy_train.rb

```diff
@@ -57,6 +62,15 @@ def setup
             super
             @round.bought_trains = false
           end
+
+          def buyable_trains(entity)
+            trains_to_buy = super
+            trains_to_buy.reject! { |t| t.name == '2P' } if entity.trains.count { |t| t.name == '2P' } == 1
+            # Can't buy trains from other corporations until train 3
+            return trains_to_buy if @game.can_buy_trains
```

Comment:
@game.can_buy_trains ? trains_to_buy : trains_to_buy.select(&:from_depot?)

Also something like can_cross_buy? would be cleared than can_buy_trains.